### ISSUE TITLE
DNS issue in containers

### ISSUE THREAD

**Issue by bonanza123** [mailcow/mailcow-dockerized#2132]
I know that there are already several other issues in this regard, but none of them seem to provide a solution for my case.

I have mailcow docker-comse version `3fa42ada8e04c79e803144632bca8923db350066` and the problem that DNS resolution is not working in all containers but unbound.
As `dig` or `nslookup` is not installed in most containers, I am using `wget` or `curl` to test it:

```
# docker-compose exec postfix-mailcow curl web.de
WARNING: The WATCHDOG_NOTIFY_EMAIL variable is not set. Defaulting to a blank string.
curl: (6) Could not resolve host: web.de
```

```
docker-compose exec sogo-mailcow wget web.de
WARNING: The WATCHDOG_NOTIFY_EMAIL variable is not set. Defaulting to a blank string.
--2018-12-27 16:47:00--  http://web.de/
Resolving web.de (web.de)... failed: Temporary failure in name resolution.
wget: unable to resolve host address 'web.de'
```

Here for unbound:
```
# docker-compose exec unbound-mailcow wget web.de
WARNING: The WATCHDOG_NOTIFY_EMAIL variable is not set. Defaulting to a blank string.
Connecting to web.de (82.165.229.138:80)
Connecting to web.de (82.165.230.17:443)
index.html           100% |********************************************|   346k  0:00:00 ETA
```

Here is dump from `iptables -S`: https://pastebin.com/PcEiiwJJ
where `172.111.1.x` is the subnet with mailcow.

P.S.: I am using hetzner but have disabled their firewall.

**Comment by andryyy** [comment]
You need to ask why this traffic is blocked.

They still have a huge IDS in their network, even when you disable their firewall.

You may also sniff the network traffic etc. 

Do IPv4 lookups work? Don’t test with wget. Use dig or drill please. Just install them with apt or apk. :)

If someone gave me a Hetzner VM I would test it.

**Comment by bonanza123** [comment]
Thanks for your response!

Do you have an idea where to start with sniffing the traffic? Inside the unbound/sogo container or on the host system? (The install process was a bid difficult without DNS). How would I do this? Just iptables logging?

Here some results with `dig`:

```
docker-compose exec unbound-mailcow dig web.de @127.0.0.1                                                                         
WARNING: The WATCHDOG_NOTIFY_EMAIL variable is not set. Defaulting to a blank string.

; <<>> DiG 9.12.3 <<>> web.de @127.0.0.1
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 27576
;; flags: qr rd ra ad; QUERY: 1, ANSWER: 2, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4096
;; QUESTION SECTION:
;web.de.                                IN      A

;; ANSWER SECTION:
web.de.                 60      IN      A       82.165.229.138
web.de.                 60      IN      A       82.165.230.17

;; Query time: 6 msec
;; SERVER: 127.0.0.1#53(127.0.0.1)
;; WHEN: Thu Dec 27 17:05:17 UTC 2018
;; MSG SIZE  rcvd: 67
```

and

```
docker-compose exec sogo-mailcow dig web.de @127.0.0.1
WARNING: The WATCHDOG_NOTIFY_EMAIL variable is not set. Defaulting to a blank string.

; <<>> DiG 9.10.3-P4-Debian <<>> web.de @127.0.0.1
;; global options: +cmd
;; connection timed out; no servers could be reached
```

**Comment by andryyy** [comment]
Start with port 53. :P

Don’t know. Ask Hetzner please. They may see spoofed traffic via port 53 and block it.

I don’t have a Hetzner machine for tests.

**Comment by bonanza123** [comment]
Thanks again for your response.

I used `tcpdump -ni eth0 udp port 53` to log DNS queries. Those from unbound are logged there successfully, while those e.g. from the sogo container do not show up at all.

In the unbound logs, I see a lot of 
```
unbound-mailcow_1    | [1545932991] unbound[1:0] debug: refused query from ip4 172.111.1.4 port 38122 (len 16)
```

I then wanted to add `172.111.1.4/24` to the subnets but only then noticed that I did something stupid as this subnet is not private as per RFC. I now reconfigured the subnet and DNS is working fine again.
